jsapp/js/components/submissions/table.es6

```diff
@@ -374,69 +482,64 @@ export class DataTable extends React.Component {
         <ValidationStatusDropdown
           onChange={this.onValidationStatusChange.bind(this, row.original._id, row.index)}
           currentValue={this.getValidationStatusOption(row.original)}
-          isDisabled={!(this.isSubmissionWritable('validate_submissions', this.props.asset, row.original))}
+          isDisabled={!(this.isSubmissionWritable(PERMISSIONS_CODENAMES.validate_submissions, this.props.asset, row.original))}
         />
       ),
     };
   }
 
   /**
-   * Prepares data for react table
-   * @param {object} data
+   * Builds and gathers all necessary react-table data and stores in state.
+   *
+   * @param {object[]} data - list of submissions
    */
   _prepColumns(data) {
-    const displayedColumns = this.getDisplayedColumns(data);
+    const allColumns = tableStore.getAllColumns(data);
 
     let showLabels = this.state.showLabels;
     let showGroupName = this.state.showGroupName;
     let showHXLTags = this.state.showHXLTags;
-    let settings = this.props.asset.settings;
     let translationIndex = this.state.translationIndex;
-    let maxPageRes = Math.min(this.state.pageSize, this.state.tableData.length);
-    let _this = this;
+    let maxPageRes = Math.min(this.state.pageSize, this.state.submissions.length);
 
-    if (settings['data-table'] && settings['data-table']['translation-index'] !== null) {
-      translationIndex = settings['data-table']['translation-index'];
-      showLabels = translationIndex > -1 ? true : false;
```

Comment:
Let me know if you can reproduce it, but for me `showLabels` is false by default. If you make a fresh form with media questions, make some submissions, then go to the data table, the XML names are being used instead

I don't know if this seemingly redundant line is responsible :D

Reply:
Clicking "Save" without doing anything in the table settings fixes this

Reply:
I tried reproducing it:

1. Make form with some questions (both with and without media questions)
2. Make submissions
3. Open Table View

I see the Labels, not XML names. Also if I open Display Options I see "Labels" selected.

Reply:
Could you try reproducing it again? Maybe the form needs to have more languages?

Reply:
Found it and fixed it :)